### ISSUE TITLE
Unlock database

### ISSUE THREAD

**Issue by andymarden** [vrtmrz/obsidian-livesync#360]
It's all very well having a Lock Database option but where is the unlock for afterwards?

If it happens automatically in some way, could this be explained in that option?

**Comment by dlphnkck** [comment]
[docs/settings.md](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/settings.md)→[Hatch](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/settings.md#miscellaneous)

> ## Hatch
> From here, everything is under the hood. Please handle it with care.
> 
> When there are problems with synchronization, the warning message is shown Under this section header.
> 
> - Pattern 1  
> ![CorruptedData](https://i.gyazo.com/52b8ac74b53be716fe3415bade3dfd7d.png)
> This message is shown when the remote database is locked and your device is not marked as "resolved".  
> Almost it is happened by enabling End-to-End encryption or History has been dropped.  
> If you enabled End-to-End encryption, you can unlock the remote database by "Apply and receive" automatically. Or "Drop and receive" when you dropped. If you want to unlock manually, click "mark this device as resolved".
> 
> - Pattern 2  
> ![Image from Gyazo](https://i.gyazo.com/3d995bb5fab49da14926f8d9dd64f8bf.png)
> The remote database indicates that has been unlocked Pattern 1.  
> When you mark all devices as resolved, you can unlock the database.
> But, there's no problem even if you leave it as it is.

If the above explanation in [settings.md](https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/settings.md) is not clear enough , please refer to the Code, TypeScript, `markRemoteLocked()`, `markRemoteUnlocked()`, and `markRemoteResolved()` sections.

This will help you understand the code of the LiveSync plugin.

**Comment by CupricReki** [comment]
I think the confusion comes from the Lock remote database button is on the Maintain database page, but the unlock database is done from the Hatch page. 

Maybe grey out the Lock button when the database is already locked and indicate somehow that the database can be unlocked elsewhere. Better yet, just combine Hatch and Maintain database pages. Them being separate is confusing.

**Comment by vrtmrz** [comment]
Sorry for being late! As you mentioned, we do not have to unlock the database.

The purpose of `Lock` is to prevent the connection of another non-empty device, and to avoid unexpected conflicts of files.  
The device whose connection is prevented will be needed to mark itself as `resolved` on the remote. (Actually,  the device which has no `resolved` mark disconnects itself).

One of the ways of marking `resolved` is @dlphnkick mentioned at first. However, as a note, it is a very manual feature and not recommended for casual usage.
Instead of that, while replicating to the rebuilt remote database (`Locked` and 
the device not `resolved`), we can decide to unlock the remote by fetching the remote again.
When fetching, the remote is marked `resolved` by the fetching device. When rebuilding a remote database, it is always locked and marked `resolved` only by the rebuilding device.

Therefore, once the remote database has been locked, it usually should not be unlocked before all your synchronised devices have been resolved. And, owning each device would not know about the past connected other devices, it could not be unlocked automatically instead of resolving.

In summary, Locking the database is a usual operation to prevent corruption. But resolving should be automatically performed and the manual operation should be under the bonnet. (And, fetching, overwriting, rebuild all is on the `Maintain database`).

I hope this provides you with the insight you were seeking!
